### ISSUE TITLE
Question: libvirt_ignition on Redhat 7.3

### ISSUE THREAD

**Issue by kpettijohn** [dmacvicar/terraform-provider-libvirt#152]
#  Version Reports:

### Distro  version of host:

``` Redhat 7.3```

### Terraform Version Report
Terraform `v0.9.11`

### Libvirt version

```
virsh version
Compiled against library: libvirt 2.0.0
Using library: libvirt 2.0.0
Using API: QEMU 2.0.0
Running hypervisor: QEMU 1.5.3
```

### terraform-provider-libvirt plugin version (git-hash)

`78341e5971cac8f23e7c36321180a71ee6816131`
# Description of Issue/Question

### Setup

Overall this is more of a question than a issue. I am attempting to use the ignition resource but it seems to rely on a feature thats not currently supported in the version of QEMU that Redhat 7.3 provides. Im curious if anyone has found a workaround in order to still use ignition files on a Redhat 7.3 host. 

Any feedback/suggestions would be much appreciated!

```
provider "libvirt" {
    uri = "qemu:///system"
}

data "ignition_user" "users" {
  count               = "${length(var.users)}"
  name                = "${lookup(var.users[count.index], "name")}"
  ssh_authorized_keys = ["${lookup(var.users[count.index], "ssh_authorized_keys")}"]
  groups              = ["sudo", "docker"]
}

data "ignition_config" "default" {
  users = [
      "${data.ignition_user.users.*.id}",
  ]
}

resource "libvirt_ignition" "coreos" {
  name = "coreos"
  pool = "container_linux"
  content = "${data.ignition_config.default.rendered}"
}

resource "libvirt_volume" "coreos_qcow2" {
  name = "coreos-qcow2"
  pool = "container_linux"
  source = "http://127.0.0.1/images/coreos_production_qemu_image.img"
}

resource "libvirt_volume" "vol_coreos_qcow2" {
  name = "vol-coreos-qcow2"
  pool = "container_linux"
  base_volume_id = "${libvirt_volume.coreos_qcow2.id}"
}

resource "libvirt_domain" "coreos" {
  name = "domain-container-host1-qcow2"
  memory = "1024"
  vcpu = 1
  disk {
    volume_id = "${libvirt_volume.vol_coreos_qcow2.id}"
  }
  coreos_ignition =  "${libvirt_ignition.coreos.id}"
}
```

### Steps to Reproduce Issue
Run a fresh Terraform apply:
```
terraform apply
data.ignition_user.users.3: Refreshing state...
data.ignition_user.users.1: Refreshing state...
data.ignition_user.users.2: Refreshing state...
data.ignition_user.users.4: Refreshing state...
data.ignition_user.users.0: Refreshing state...
data.ignition_config.default: Refreshing state...
libvirt_ignition.coreos: Refreshing state... (ID: /data/vm-images/container_linux/coreos;597f9514-a820-9970-5e9b-d2dd324fcb80)
libvirt_volume.coreos_qcow2: Refreshing state... (ID: /data/vm-images/container_linux/coreos-qcow2)
libvirt_volume.vol_coreos_qcow2: Refreshing state... (ID: /data/vm-images/container_linux/vol-coreos-qcow2)
libvirt_domain.coreos: Creating...
  coreos_ignition:  "" => "/data/vm-images/container_linux/coreos;597f9514-a820-9970-5e9b-d2dd324fcb80"
  disk.#:           "" => "1"
  disk.0.%:         "" => "1"
  disk.0.volume_id: "" => "/data/vm-images/container_linux/vol-coreos-qcow2"
  memory:           "" => "1024"
  name:             "" => "domain-container-host1-qcow2"
  running:          "" => "true"
  vcpu:             "" => "1"
Error applying plan:

1 error(s) occurred:

* libvirt_domain.coreos: 1 error(s) occurred:

* libvirt_domain.coreos: Error creating libvirt domain: virError(Code=1, Domain=10, Message='internal error: process exited while connecting to monitor: qemu-kvm: -fw_cfg: invalid option')
```

# Additional Infos:

```
getenforce
Disabled
```

Please let me know if I can provide any additional details!

**Comment by dmacvicar** [comment]
Ignition files are implemented in terraform-provider-libvirt using the [QEMU Firmware Configuration Device](https://git.qemu.org/?p=qemu.git;a=blob;f=docs/specs/fw_cfg.txt).

Ignition, supports [various sources](https://github.com/coreos/ignition/blob/master/doc/supported-platforms.md) to read this configuration. As we are using mostly QEMU with libvirt, it makes sense to use this interface to pass the data to QEMU once it is read from a helper persistent volume it is created.

In theory, there are other sources, like passing a url in the kernel command line. However, we have no way to pass this from the terraform resources into this url. Ignition does not support passing a device into the kernel command line, like [device://](https://doc.opensuse.org/projects/autoyast/#Commandline.ay) in AutoYaST, which would allow a trick like we do with `cloud-init` as well.

So in short, you need a qemu recent enough to support the `fw_cfg` option. Looks like Redhat 7.3 ships a older QEMU.

Does that answer your question?

**Comment by kpettijohn** [comment]
It sure does, thanks for taking the time and providing detailed response! Looking forward to using the provider more in the future!
